rules/aip0192/aip0192.go

```diff
@@ -35,6 +37,26 @@ func AddRules(r lint.RuleRegistry) error {
 	)
 }
 
+// Returns true if this is a deprecated method, false otherwise.
+func isDeprecated(d desc.Descriptor) bool {
+	switch d.(type) {
+	case *desc.MethodDescriptor:
+		m := d.(*desc.MethodDescriptor)
+		if m.GetMethodOptions() == nil {
+			return false
+		}
+		return *m.GetMethodOptions().Deprecated
+	case *desc.ServiceDescriptor:
+		s := d.(*desc.ServiceDescriptor)
+		if s.GetServiceOptions() == nil {
+			return false
+		}
+		return *s.GetServiceOptions().Deprecated
```

Comment:
```suggestion
	switch d := d.(type) {
	case *desc.MethodDescriptor:
		return d.GetMethodOptions().GetDeprecated()
	case *desc.ServiceDescriptor:
		return d.GetServiceOptions().GetDeprecated()
```

Reply:
This fails with the error message below. The `Options() == nil` check is also necessary so that the valid case (without `deprecated = true`) won't result in a null pointer deference error.

Error message: `d.GetMethodOptions undefined (type desc.Descriptor has no field or method GetMethodOptions)`

Reply:
I updated my suggestion to fix the error. But the options check isn't necessary because the generated Go code already handles the `nil` case.

Reply:
Thanks!